index.js

```diff
@@ -218,13 +235,24 @@ if (require.main !== module) {
     }
   });
 
-  wait_dom.then(() => {
-    load_placements()
-      .then((placements) => {
-        // Any post processing on placement list can go here
-      })
-      .catch((err) => {
-        console.error(err);
-      });
+  const is_debug = global.debug || window.debug || false;
+
+  wait = new Promise((resolve) => {
+    wait_dom.then(() => {
+      load_placements()
+        .then((placements) => {
+          resolve(placements);
+        })
+        .catch((err) => {
+          resolve([]);
+          if (err instanceof Error) {
+            if (err instanceof EthicalAdsWarning && !is_debug) {
```

Comment:
Shouldn't we just invert this logic to have the console.log in the `if` and no `else`?

Reply:
Inverting would be a bit gnarlier, as this is the one exception case. The `if` here could `return` though, and the `console.log` can be moved out of the `else` clause though.